cumulusci/tasks/bulkdata/mapping_parser.py

```diff
@@ -0,0 +1,34 @@
+from typing import Dict, List
+
+from pydantic import Field
+
+from cumulusci.utils.yaml.model_parser import CCIDictModel
+
+
+class Lookup(CCIDictModel):
+    table: str
+    key_field: str = None
+    value_field: str = None
+    join_field: str = None
+    after: str = None
+    aliased_table: str = None
+
+
+class Step(CCIDictModel):
+    sf_object: str
+    table: str = None
+    fields_: Dict[str, str] = Field(..., alias="fields")
+    lookups: Dict[str, Lookup] = {}
+    static: Dict[str, str] = {}
+    filters: List[str] = []
+    action: str = "insert"
+    oid_as_pk: bool = False  # this one should be discussed and probably deprecated
+    record_type: str = None  # should be discussed and probably deprecated
```

Comment:
`record_type` should be deprecated because we have much superior record type mapping support now. `record_type` encourages ways of structuring the data set that end up being limiting (i.e. splitting a single sobject into multiple tables on record type).

Reply:
But we need to retain support for old mappings that use it.

Reply:
I can output a warning for it if we get consensus on that.

Reply:
What should be the deprecation message? Is there any documentation on the new record type support? Is it totally automatic?

Reply:
Does this work?

            "record_type is deprecated. Just supply an RecordTypeId column declaration and it will be inferred"

Reply:
d599001a18913aa